### ISSUE TITLE
Grafana Variable dependence check failure

### ISSUE THREAD

**Issue by stevecalvert** [grafana/grafana#13600]
### What Grafana version are you using?
Grafana v5.2.4 (0bbac5c)

### What datasource are you using?
SimpleJson (SJ)

### What OS are you running grafana on?
Ubuntu 17.03

### What did you do?
Created a dashboard variable, var2, dependent on variable, var1

### What was the expected result?
var1 would be evaluated by calling the SJ /query endpoint associated with var1 before the response was used as the parameter to the SJ query associated with var2

### What happened instead?
The detection of the dependence seems to only function if the syntax for the query is
```query2 $var1```
or
```query2 [[var1]]```
 
Using the following syntax examples (which is documented on the [Grafana Variables page](http://docs.grafana.org/reference/templating/#advanced-formatting-options), and necessary for passing multiple values to the dependent variable, the test for a dependent variable appears to fail

```query2 ${var1}```
or
```query2 ${var1:csv}```
or any expression which has a ':' followed by the glob/regex/pipe/csv

Looking at the Grafana code, in ```variable.ts```, function ```containsVariable()```, it looks like the Regex evaluation does not accomodate either the ${var} syntax or any of the multi-select options that require the ':' syntax, including. for example,  ```query2 $var1:csv```

Regex example [here](https://regex101.com/r/LBB4Uu/1/)

I'm not sure if this is intended behaviour or whether I should be structuring my queries differently?

**Comment by torkelo** [comment]
thanks for reporting this! It's due to the getVariableName function not checking the 3rd match (syntax),
 
https://github.com/grafana/grafana/blob/master/public/app/features/templating/template_srv.ts#L147

**Comment by cored** [comment]
@torkelo Hi, I'm interested in working on this issue.

**Comment by marefr** [comment]
@cored awesome. Please reach out if you need help with anything

**Comment by cored** [comment]
thanks @marefr will do!

**Comment by marefr** [comment]
Hey @cored are you still looking into this or not?

**Comment by cored** [comment]
Will work on it this afternoon, yes.

**Comment by marefr** [comment]
@cored Thanks. We would like to include fix for this in upcoming 5.3.1 release, currently planned tomorrow (Tuesday) or Wednesday. For us to review and making possible changes we would need a PR asap tomorrow, but no pressure :smile:

**Comment by cored** [comment]
Hehehe, cool.

**Comment by cored** [comment]
Hi @maref I don't think I will be able to make it; I didn't notice the milestone before starting the issue and also didn't notice that this was a frontend problem. I was trying to get grafana going locally without any luck and without testing I don't feel confident to open a PR. 

Getting the following error after following the instructions on the README: 
```
 Error occurred while trying to proxy request / from localhost:3333 to http://localhost:3000 (ECONNREFUSED) (https://nodejs.org/api/errors.html#errors_common_system_errors)
```
I'll keep trying to get the development environment going, however, it will be to help out with a different issue. Sorry for the delay reporting this, I just had time today to work on the ticket.

**Comment by marefr** [comment]
@cored sorry wasn't meant to stress you. We've postponed this to a later milestone. Please let me know if you're still interested in this.

> Getting the following error after following the instructions on the README:
> 
> ```
>  Error occurred while trying to proxy request / from localhost:3333 to http://localhost:3000 (ECONNREFUSED) (https://nodejs.org/api/errors.html#errors_common_system_errors)
> ```
> I'll keep trying to get the development environment going, however, it will be to help out with a different issue. Sorry for the delay reporting this, I just had time today to work on the ticket.

Guessing you're on Windows, correct? Think the ports 3000 and 3333 are problematic there and needs some extra permissions - therefore changing the port to 8080 or similar [may be needed](http://docs.grafana.org/installation/windows/#configure). Otherwise you don't have the Grafana backend running?

An alternative to `yarn start` would be `yarn watch` which don't use webpack development server and relies on Grafana backend running on localhost:3000.

**Comment by cored** [comment]
Oh sounds good, I'll like to keep going then. No I'm using OSX

**Comment by marefr** [comment]
Okay. Then maybe you have something else running on port 3000? You're sure that you have grafana backend running? Please try `yarn watch` instead.

**Comment by marefr** [comment]
@cored if you need any help please join our public slack at http://slack.raintank.io/. You can find me as `mefraimsson` there.

**Comment by cored** [comment]
Sure thing, I wasn't able to take a look into this today. I'll make sure to
work on it tomorrow.

**Comment by cored** [comment]
@marefr Yes, sorry for the delay. I got the local environment running yesterday night; however, wasn't able to replicate the issue. I will check the issue list and see where I can contribute to.

**Comment by jschill** [comment]
Feel free to try if the PR solves this issue.
